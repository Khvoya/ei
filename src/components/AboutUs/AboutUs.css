.AboutUs {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.AboutUs__slider-wrap {
    padding: 0;
}

.AboutUs__text {
    background-color: white;
    padding: 4vw 8vw;
    box-sizing: border-box;
}

.AboutUs__text h1 {
    text-align: center;
}

.AboutUs__text p {
    font-weight: 400;
    text-align: start;
    padding-top: 20px;
    font-size: 20px;
    line-height: 26px;
}

.carousel, .carousel-inner, .carousel-item , .carousel-item img{
    width: 100%;
    height: 100%;
    min-height: 375px;
    object-fit: cover;
}
@media screen and (max-width: 960px) {
    .AboutUs__wrap {
        margin: 0;
    }
}
@media screen and (max-width: 420px) {
    .AboutUs__text p {
        font-size: 16px;
    }
}
@media screen and (min-width: 1400px) {
    .AboutUs__text p {
        line-height: 2.5vw;
        font-size: 25px;
    }
}
