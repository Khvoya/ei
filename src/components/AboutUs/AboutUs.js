import React, { Component } from "react";
import { Carousel, Col } from "react-bootstrap";
import "./AboutUs.css";
import {connect} from 'react-redux';

class AboutUs extends Component {
  render() {
    const { urls } = this.props;
    const photos = urls.map((item, index) => (
      <Carousel.Item key={index}>
        <img className="d-block w-100" alt="slide" src={item} />
      </Carousel.Item>
    ));
    return (
      <div id="band" className="AboutUs">
        <Col
          className="AboutUs__slider-wrap"
          xl="6"
          lg="12"
          md="12"
          sm="12"
          xs="12"
        >
          <Carousel
            controls={false}
            fade={true}
            pauseOnHover={false}
            interval={10000}
          >
            {photos}
          </Carousel>
        </Col>
        <Col xl="6" lg="12" md="12" sm="12" xs="12" className="AboutUs__text">
          <h1>Band</h1>
          <p>
            In ordinary life we are three engineers and teacher but when we get
            together- we are "Electric Indians". Despite the fact that this
            project was formed in 2016, each of us has the experience of playing
            in groups of different directions from electronic music to metal,
            which we unite in our music using an ambient transitions, african
            and jungle-beats, vocal performances with processor and looper etc.
            It's not capable of "sticking to the shore" of a particular style:
            the mood, the shade of genres and even language can vary from song
            to song withiout losing the spirit of rock and metal schools, of
            course. Also, using the work with lighting devices, any of our shows
            turns into a kind of "ritual", which will not leave indifferent any
            participant of the event..
          </p>
        </Col>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  urls: state.aboutUs.urls,
});
export default connect(mapStateToProps)(AboutUs);
