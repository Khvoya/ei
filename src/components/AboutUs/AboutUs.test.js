import React from "react";
import { shallow } from "enzyme";
import { AboutUs } from "./AboutUs";

describe("AboutUs component", () => {
  const props = {
    urls: [
      "https://first-test-url.com",
      "https://second-test-url.com",
      "https://third-test-url.com"
    ]
  };

  it("should render correctly", () => {
    const output = shallow(<AboutUs {...props} />);
    expect(output).toMatchSnapshot();
  });
});
