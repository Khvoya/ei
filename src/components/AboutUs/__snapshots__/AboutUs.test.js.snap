// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutUs
    urls={
      Array [
        "https://first-test-url.com",
        "https://second-test-url.com",
        "https://third-test-url.com",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef
          as="div"
          className="AboutUs__slider-wrap"
          lg="12"
          md="12"
          sm="12"
          xl="6"
          xs="12"
        >
          <ForwardRef(render)
            controls={false}
            fade={true}
            interval={8000}
            pauseOnHover={false}
            slide={true}
          >
            <ForwardRef>
              <img
                alt="slide"
                className="d-block w-100"
                src="https://first-test-url.com"
              />
            </ForwardRef>
            <ForwardRef>
              <img
                alt="slide"
                className="d-block w-100"
                src="https://second-test-url.com"
              />
            </ForwardRef>
            <ForwardRef>
              <img
                alt="slide"
                className="d-block w-100"
                src="https://third-test-url.com"
              />
            </ForwardRef>
          </ForwardRef(render)>
        </ForwardRef>,
        <ForwardRef
          as="div"
          className="AboutUs__text"
          lg="12"
          md="12"
          sm="12"
          xl="6"
          xs="12"
        >
          <h1>
            Band
          </h1>
          <p>
            In ordinary life we are three engineers and teacher but when we get together- we are "Electric Indians". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of "sticking to the shore" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of "ritual", which will not leave indifferent any participant of the event..
          </p>
        </ForwardRef>,
      ],
      "className": "AboutUs",
      "id": "band",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <ForwardRef(render)
            controls={false}
            fade={true}
            interval={8000}
            pauseOnHover={false}
            slide={true}
          >
            <ForwardRef>
              <img
                alt="slide"
                className="d-block w-100"
                src="https://first-test-url.com"
              />
            </ForwardRef>
            <ForwardRef>
              <img
                alt="slide"
                className="d-block w-100"
                src="https://second-test-url.com"
              />
            </ForwardRef>
            <ForwardRef>
              <img
                alt="slide"
                className="d-block w-100"
                src="https://third-test-url.com"
              />
            </ForwardRef>
          </ForwardRef(render)>,
          "className": "AboutUs__slider-wrap",
          "lg": "12",
          "md": "12",
          "sm": "12",
          "xl": "6",
          "xs": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://first-test-url.com"
                />
              </ForwardRef>,
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://second-test-url.com"
                />
              </ForwardRef>,
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://third-test-url.com"
                />
              </ForwardRef>,
            ],
            "controls": false,
            "fade": true,
            "interval": 8000,
            "pauseOnHover": false,
            "slide": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "children": <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://first-test-url.com"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "slide",
                  "className": "d-block w-100",
                  "src": "https://first-test-url.com",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "displayName": "CarouselItem",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://second-test-url.com"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "slide",
                  "className": "d-block w-100",
                  "src": "https://second-test-url.com",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "displayName": "CarouselItem",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "children": <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://third-test-url.com"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "slide",
                  "className": "d-block w-100",
                  "src": "https://third-test-url.com",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "displayName": "CarouselItem",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Caption": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "CarouselCaption",
              "render": [Function],
            },
            "Item": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "CarouselItem",
              "render": [Function],
            },
            "defaultProps": undefined,
            "displayName": "Bootstrap(undefined)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": Array [
            <h1>
              Band
            </h1>,
            <p>
              In ordinary life we are three engineers and teacher but when we get together- we are "Electric Indians". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of "sticking to the shore" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of "ritual", which will not leave indifferent any participant of the event..
            </p>,
          ],
          "className": "AboutUs__text",
          "lg": "12",
          "md": "12",
          "sm": "12",
          "xl": "6",
          "xs": "12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Band",
            },
            "ref": null,
            "rendered": "Band",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "In ordinary life we are three engineers and teacher but when we get together- we are \\"Electric Indians\\". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of \\"sticking to the shore\\" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of \\"ritual\\", which will not leave indifferent any participant of the event..",
            },
            "ref": null,
            "rendered": "In ordinary life we are three engineers and teacher but when we get together- we are \\"Electric Indians\\". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of \\"sticking to the shore\\" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of \\"ritual\\", which will not leave indifferent any participant of the event..",
            "type": "p",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef
            as="div"
            className="AboutUs__slider-wrap"
            lg="12"
            md="12"
            sm="12"
            xl="6"
            xs="12"
          >
            <ForwardRef(render)
              controls={false}
              fade={true}
              interval={8000}
              pauseOnHover={false}
              slide={true}
            >
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://first-test-url.com"
                />
              </ForwardRef>
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://second-test-url.com"
                />
              </ForwardRef>
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://third-test-url.com"
                />
              </ForwardRef>
            </ForwardRef(render)>
          </ForwardRef>,
          <ForwardRef
            as="div"
            className="AboutUs__text"
            lg="12"
            md="12"
            sm="12"
            xl="6"
            xs="12"
          >
            <h1>
              Band
            </h1>
            <p>
              In ordinary life we are three engineers and teacher but when we get together- we are "Electric Indians". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of "sticking to the shore" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of "ritual", which will not leave indifferent any participant of the event..
            </p>
          </ForwardRef>,
        ],
        "className": "AboutUs",
        "id": "band",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <ForwardRef(render)
              controls={false}
              fade={true}
              interval={8000}
              pauseOnHover={false}
              slide={true}
            >
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://first-test-url.com"
                />
              </ForwardRef>
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://second-test-url.com"
                />
              </ForwardRef>
              <ForwardRef>
                <img
                  alt="slide"
                  className="d-block w-100"
                  src="https://third-test-url.com"
                />
              </ForwardRef>
            </ForwardRef(render)>,
            "className": "AboutUs__slider-wrap",
            "lg": "12",
            "md": "12",
            "sm": "12",
            "xl": "6",
            "xs": "12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef>
                  <img
                    alt="slide"
                    className="d-block w-100"
                    src="https://first-test-url.com"
                  />
                </ForwardRef>,
                <ForwardRef>
                  <img
                    alt="slide"
                    className="d-block w-100"
                    src="https://second-test-url.com"
                  />
                </ForwardRef>,
                <ForwardRef>
                  <img
                    alt="slide"
                    className="d-block w-100"
                    src="https://third-test-url.com"
                  />
                </ForwardRef>,
              ],
              "controls": false,
              "fade": true,
              "interval": 8000,
              "pauseOnHover": false,
              "slide": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "children": <img
                    alt="slide"
                    className="d-block w-100"
                    src="https://first-test-url.com"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "slide",
                    "className": "d-block w-100",
                    "src": "https://first-test-url.com",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "displayName": "CarouselItem",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "children": <img
                    alt="slide"
                    className="d-block w-100"
                    src="https://second-test-url.com"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "slide",
                    "className": "d-block w-100",
                    "src": "https://second-test-url.com",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "displayName": "CarouselItem",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "function",
                "props": Object {
                  "children": <img
                    alt="slide"
                    className="d-block w-100"
                    src="https://third-test-url.com"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "slide",
                    "className": "d-block w-100",
                    "src": "https://third-test-url.com",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "displayName": "CarouselItem",
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Caption": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "displayName": "CarouselCaption",
                "render": [Function],
              },
              "Item": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "displayName": "CarouselItem",
                "render": [Function],
              },
              "defaultProps": undefined,
              "displayName": "Bootstrap(undefined)",
              "propTypes": undefined,
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": Array [
              <h1>
                Band
              </h1>,
              <p>
                In ordinary life we are three engineers and teacher but when we get together- we are "Electric Indians". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of "sticking to the shore" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of "ritual", which will not leave indifferent any participant of the event..
              </p>,
            ],
            "className": "AboutUs__text",
            "lg": "12",
            "md": "12",
            "sm": "12",
            "xl": "6",
            "xs": "12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Band",
              },
              "ref": null,
              "rendered": "Band",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "In ordinary life we are three engineers and teacher but when we get together- we are \\"Electric Indians\\". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of \\"sticking to the shore\\" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of \\"ritual\\", which will not leave indifferent any participant of the event..",
              },
              "ref": null,
              "rendered": "In ordinary life we are three engineers and teacher but when we get together- we are \\"Electric Indians\\". Despite the fact that this project was formed in 2016, each of us has the experience of playing in groups of different directions from electronic music to metal, which we unite in our music using an ambient transitions, african and jungle-beats, vocal performances with processor and looper etc. It's not capable of \\"sticking to the shore\\" of a particular style: the mood, the shade of genres and even language can vary from song to song withiout losing the spirit of rock and metal schools, of course. Also, using the work with lighting devices, any of our shows turns into a kind of \\"ritual\\", which will not leave indifferent any participant of the event..",
              "type": "p",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
